### ISSUE TITLE
Feature: Test 'Backup destination' connection

### ISSUE THREAD

**Issue by PitWenkin** [virtualmin/virtualmin-gpl#279]
When creating/editing a scheduled backup there is no possibility to check if the connection settings are actually working.
To check if they are one has to execute the backup, which might then take some time to complete.

As far as I know when starting a backup, the connection is checked, and if it fails the backup is canceled. Thus there seems to exist a connection check, wich should be able to be trigger via a button?

![Screenshot 2021-04-15 at 11 44 34](https://user-images.githubusercontent.com/45786226/114849683-49b89200-9de0-11eb-82f7-ffdf9709e33f.png)

**Comment by jcameron** [comment]
Not a bad idea ... although after creating a scheduled backup you can trigger it immediately from the UI as a way of testing the destination.

**Comment by PitWenkin** [comment]
Even triggering the backup will start the whole creation of tar files and do the actual transfer of data, which would not be necessary.

**Comment by jcameron** [comment]
The backup process will test the remote connection before creating the tar file though, so it will fail early if the destination is invalid.

**Comment by chris001** [comment]
True. But if the credentials are succeed, then it'll start the backup process which might take hours or days to complete!  Don't want it to start the backup process, just test the credentials for the connection are succeed or fail.

**Comment by iliarostovtsev** [comment]
> Thus there seems to exist a connection check, wich should be able to be trigger via a button?

Jamie, we could do it for multiple pages in a single API (to test connections), using a popup if you wish, the same way as for upload progress, and I could transform it into a nicer, non-popup connection tester on Authentic Theme?

**Comment by jcameron** [comment]
Yes, a popup could work. Alternately we could have a button to test a scheduled backup without actually running the full process?

**Comment by iliarostovtsev** [comment]
> Alternately we could have a button to test a scheduled backup without actually running the full process?

That would be great as well. Moreover, it would be an interesting feature to test connection before any forms are saved. We have a lot of pages where you need to setup a connection (FTP, SSH, Webmin Server, Amazon and etc). I suggest to add a button as on the screenshot below to test the connection completely independently:

![image](https://user-images.githubusercontent.com/4426533/115527903-2dae6800-a29a-11eb-8f38-7ecc2afcd351.png)

**Comment by PitWenkin** [comment]
Any news on this?

**Comment by iliajie** [comment]
No, we never implemented it, at least not yet.

**Comment by jcameron** [comment]
Closest we've gotten is to test this at the start of a backup. I suppose we could add a test button on the list of scheduled backups?

**Comment by iliajie** [comment]
Yes, we can certainly add a button for testing a connection. Though, I could only do it a bit later, after I finish Virtualmin website re-work..

**Comment by iliaross** [comment]
❗️ If the issue isn't resolved, feel free to reopen it with updated details.

**Comment by PitWenkin** [comment]
As far as i know it was never implemented. But i dont really need it ;-)

**Comment by iliaross** [comment]
Yeah, it's better to keep it simple since the next screen after submitting the form would show an error anyway.
